### PR TITLE
fix(lint): enable type checker

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes NAN-454 

- Enable Typescript type checking linter
It makes the linter a little slower but catch way more issues. 

- Ran `--fix` on the packages 
It mostly removed unnecessary typings

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,37 +2,53 @@
...
-      // Previously warn but now errors
-      "@typescript-eslint/no-non-null-assertion": "warn",
-      "@typescript-eslint/no-explicit-any": "warn"
+        "no-console": 1, // Means warning
+        "prettier/prettier": 2,
+        "@typescript-eslint/no-inferrable-types": "off",
+        // Previously warn but now errors
+        "@typescript-eslint/no-non-null-assertion": "warn",
+        "@typescript-eslint/no-explicit-any": "warn",
+        // To reenable as error progressively
```
Comment (bodinsamuel) [inline .eslintrc:36]: To avoid adding errors everywhere I put those as warn, but those are relevant errors that we want to fix

**Comment by bodinsamuel** [comment]
Can I have a final review on this please? @khaliqgant @TBonnin  ☺️

**Review by khaliqgant — APPROVED** [review]
Looks good, shouldn't fix be run regularly -- aka part of the build or precommit phase?

**Comment by bodinsamuel** [comment]
> Looks good, shouldn't fix be run regularly -- aka part of the build or precommit phase?

We could at precommit like we do with prettier, I think it's also up to each developer to configure their IDE so that eslint runs along the edition. And at some point we will enforce eslint in the CI (hopefully)

File: .eslintrc [hunk 2]
```diff
@@ -2,37 +2,53 @@
...
+        "@typescript-eslint/require-await": "warn",
+        "@typescript-eslint/no-floating-promises": "warn",
```
Comment (TBonnin) [inline .eslintrc:38]: ❤️ ❤️ ❤️

**Pull request merged**